### PR TITLE
Adding Footer and Front-End Improvements

### PR DESCRIPTION
**Pull request by Ajay-Dhangar**
## Change Summary

https://github.com/wingkwong/leetcode-the-hard-way/issues/652

- [x] Navbar
- [ ] Footer
- [ ] About section

## Checklist

**If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed and will be closed without notice. Regular contributors (with 10+ PRs) can skip this part.**

### General 

- [x] This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)
- [x] I've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md)
- [x] I've applied LaTex for all variables, formulas and time / space complexity instead of using backticks
- [x] I've started the app locally and verified all the content and all links (if applicable) are accessible correctly
- [x] I've included Complexity Analysis (Time Complexity & Space Complexity).
- [x] I've written my explanation well and it is easy to understand for beginners

### Tutorial

- [ ] I've read and followed the [Tutorial Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorial-template)
- [ ] I've explained my topic well with 2 - 3 LC problems and no external problems are used. 
- [ ] I've provided the full working solutions to the problems used in this tutorial.
- [ ] I've provided suggested problems at the end with the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] I've given credits / references if I use external resources. (For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).)

### Solutions

- [ ] I've read and followed the [Solution Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#solution-template)
- [x] I've formatted my code well with [K&R Coding style](https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr)
- [ ] I've confirmed that comments are put above each line rather than writing on the same line.
- [ ] I've included a meaningful approach name for my solution. e.g. `## Approach 1: Two Pointers`.

### CODE REVIEW

**Review by Ajay-Dhangar — COMMENTED** [review]
Adding code for Footer, Navbar, about section and Front-End Improvements with responsive way

**Review by Ajay-Dhangar — COMMENTED** [review]
screencapture-localhost-3000-2023-10-02-21_27_26

**Review by Ajay-Dhangar — COMMENTED** [review]
ok I do to work but i want to say to you can remove emoji's.

**Comment by wingkwong** [comment]
@Ajay-Dhangar not a good solution.

**Review by Ajay-Dhangar — COMMENTED** [review]
updated

File: docusaurus.config.js [hunk 1]
```diff
@@ -135,146 +135,164 @@ const config = {
           },
         ],
       },
-      footer: {
-        style: 'dark',
-        links: [
+footer: {
```
Comment (wingkwong) [inline docusaurus.config.js:138]: please fix the indentation

**Review by wingkwong — CHANGES_REQUESTED** [review]
please remove those unused icon images.

**Review by Ajay-Dhangar — COMMENTED** [review]
`docusaurus.config.js`

```js
items: [
          {
            html: '<span class="nav-emoji">📚</span> Tutorials',
            to: '/tutorials/category/basic-topics',
            activeBaseRegex: `/tutorials/`,
          },
          {
            html: '<span class="nav-emoji">📖</span> Solutions',
            to: '/solutions/category/0000---0099',
            activeBaseRegex: `/solutions/`,
          },
          {
            html: '<span class="nav-emoji">🏷️</span> Tags',
            to: '/solutions/tags/',
            activeBaseRegex: `/solutions/tags/`,
          },
          {
            html: '<span class="nav-emoji">🏛️</span> Roadmap',
            to: '/roadmap/',
            activeBaseRegex: `/roadmap/`,
          },
          {
            html: '<span class="nav-emoji">🖥️</span> Templates',
            to: '/templates/',
            activeBaseRegex: `/templates/`,
          },
          // {to: '/blog', label: 'Blog', position: 'left'},
          {
            href: 'https://github.com/wingkwong/leetcode-the-hard-way',
            position: 'right',
            className: 'header-github-link',
            'aria-label': 'GitHub repository',
          },
          {
            href: 'https://discord.com/invite/Nqm4jJcyBf',
            position: 'right',
            className: 'navbar-icon',
            'aria-label': 'Discord channel',
            html: `<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 640 512" height="24" width="24" xmlns="http://www.w3.org/2000/svg"><path d="M524.531,69.836a1.5,1.5,0,0,0-.764-.7A485.065,485.065,0,0,0,404.081,32.03a1.816,1.816,0,0,0-1.923.91,337.461,337.461,0,0,0-14.9,30.6,447.848,447.848,0,0,0-134.426,0,309.541,309.541,0,0,0-15.135-30.6,1.89,1.89,0,0,0-1.924-.91A483.689,483.689,0,0,0,116.085,69.137a1.712,1.712,0,0,0-.788.676C39.068,183.651,18.186,294.69,28.43,404.354a2.016,2.016,0,0,0,.765,1.375A487.666,487.666,0,0,0,176.02,479.918a1.9,1.9,0,0,0,2.063-.676A348.2,348.2,0,0,0,208.12,430.4a1.86,1.86,0,0,0-1.019-2.588,321.173,321.173,0,0,1-45.868-21.853,1.885,1.885,0,0,1-.185-3.126c3.082-2.309,6.166-4.711,9.109-7.137a1.819,1.819,0,0,1,1.9-.256c96.229,43.917,200.41,43.917,295.5,0a1.812,1.812,0,0,1,1.924.233c2.944,2.426,6.027,4.851,9.132,7.16a1.884,1.884,0,0,1-.162,3.126,301.407,301.407,0,0,1-45.89,21.83,1.875,1.875,0,0,0-1,2.611,391.055,391.055,0,0,0,30.014,48.815,1.864,1.864,0,0,0,2.063.7A486.048,486.048,0,0,0,610.7,405.729a1.882,1.882,0,0,0,.765-1.352C623.729,277.594,590.933,167.465,524.531,69.836ZM222.491,337.58c-28.972,0-52.844-26.587-52.844-59.239S193.056,219.1,222.491,219.1c29.665,0,53.306,26.82,52.843,59.239C275.334,310.993,251.924,337.58,222.491,337.58Zm195.38,0c-28.971,0-52.843-26.587-52.843-59.239S388.437,219.1,417.871,219.1c29.667,0,53.307,26.82,52.844,59.239C470.715,310.993,447.538,337.58,417.871,337.58Z"></path></svg>`,
          },
          {
            href: 'https://www.youtube.com/@leetcodethehardway?sub_confirmation=1',
            position: 'right',
            className: 'navbar-icon',
            'aria-label': 'YouTube channel',
            html: `<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="currentColor" viewBox="0 0 16 16">
                    <path d="M8.051 1.999h.089c.822.003 4.987.033 6.11.335a2.01 2.01 0 0 1 1.415 1.42c.101.38.172.883.22 1.402l.01.104.022.26.008.104c.065.914.073 1.77.074 1.957v.075c-.001.194-.01 1.108-.082 2.06l-.008.105-.009.104c-.05.572-.124 1.14-.235 1.558a2.007 2.007 0 0 1-1.415 1.42c-1.16.312-5.569.334-6.18.335h-.142c-.309 0-1.587-.006-2.927-.052l-.17-.006-.087-.004-.171-.007-.171-.007c-1.11-.049-2.167-.128-2.654-.26a2.007 2.007 0 0 1-1.415-1.419c-.111-.417-.185-.986-.235-1.558L.09 9.82l-.008-.104A31.4 31.4 0 0 1 0 7.68v-.123c.002-.215.01-.958.064-1.778l.007-.103.003-.052.008-.104.022-.26.01-.104c.048-.519.119-1.023.22-1.402a2.007 2.007 0 0 1 1.415-1.42c.487-.13 1.544-.21 2.654-.26l.17-.007.172-.006.086-.003.171-.007A99.788 99.788 0 0 1 7.858 2h.193zM6.4 5.209v4.818l4.157-2.408L6.4 5.209z"/>
                  </svg>`,
          },
        ],
```

and `css`

```css
@media screen and (min-width: 1000px) and (max-width: 1119px) {
  .nav-emoji{
    display: none;
  }
}
```

File: docusaurus.config.js (hunk 1 above)
Comment (Ajay-Dhangar) [reply docusaurus.config.js:138]: Okay

Comment (Ajay-Dhangar) [reply docusaurus.config.js]: Can you want change footer or not I don't changed it only change navbar for responsive size /  device but I don't have more links for your footer so, i don't to try for footer but navbar fully working for responsive devices.

**Review by Ajay-Dhangar — COMMENTED** [review]
All is okay to my side can you review and give me your response.

**Comment by Ajay-Dhangar** [comment]
can you check out

File: src/css/custom.css [hunk 2]
```diff
@@ -225,4 +225,12 @@ table, tr, td, th {
 
 .hard {
   color: rgb(255 45 85);
+}
+
+/*  for emoji's */
+
+@media screen and (min-width: 1000px) and (max-width: 1119px) {
```
Comment (wingkwong) [inline src/css/custom.css:232]: mobile first principles please. no max-width.

**Review by wingkwong — CHANGES_REQUESTED** [review]
You should always verify it first.

image

**Comment by Ajay-Dhangar** [comment]
> please remove those unused icon images.

I removed those unused icon images.

File: src/css/custom.css (hunk 2 above)
Comment (Ajay-Dhangar) [reply src/css/custom.css]:
```css
  @media screen and (min-width: 997px) and (max-width: 1119px) {
    .nav-emoji{
       display: none;
    }
  }
  ```

Comment (Ajay-Dhangar) [reply src/css/custom.css]: I used for between size (997px - 1119px) only not for other

**Comment by Ajay-Dhangar** [comment]
> You should always verify it first.
> 
> image

**Size (997px - 1119px)**

image

image

**Size (1120px)**

image

**Size (996px)**

image

image

File: src/css/custom.css [hunk 3]
```diff
@@ -225,4 +225,12 @@ table, tr, td, th {
 
 .hard {
   color: rgb(255 45 85);
+}
+
+/*  for emoji's */
+
+@media screen and (min-width: 997px) and (max-width: 1119px) {
```
Comment (wingkwong) [inline src/css/custom.css:232]: like I said before. don't use between and max-width. Please follow mobile first principle. (i.e. by default no emoji is shown, then show only when min-width is 1200px.

**Review by wingkwong — COMMENTED** [review]
please fix this as well.

image

File: src/css/custom.css (hunk 3 above)
Comment (Ajay-Dhangar) [reply src/css/custom.css:232]: okay, I understand, I do to resolve this.

**Review by Ajay-Dhangar — COMMENTED** [review]
Completed For **AnnoucementBar**

image

**Comment by Ajay-Dhangar** [comment]
Can you review now,

File: src/css/custom.css (hunk 3 above)
Comment (wingkwong) [reply src/css/custom.css:234]: Not resolved at all...

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: src/css/custom.css (hunk 2 above)
Comment (Ajay-Dhangar) [reply src/css/custom.css]:
```css
@media screen and (min-width: 997px) {
  .nav-emoji{
    display: none;
  }
}
@media screen and (min-width: 1120px) {
  .nav-emoji{
    display: inline-block;
  }
}
```

**Review by Ajay-Dhangar — COMMENTED** [review]
Now, Check it only for min-width (❌ max-width).

File: src/css/custom.css [hunk 4]
```diff
@@ -231,8 +231,13 @@ table, tr, td, th {
 
 /*  for emoji's */
 
-@media screen and (min-width: 997px) and (max-width: 1119px) {
+@media screen and (min-width: 997px) {
```
Comment (wingkwong) [inline src/css/custom.css:234]: by default it is `display: none`, then you don't need `@media` at all.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: src/css/custom.css (hunk 4 above)
Comment (Ajay-Dhangar) [reply src/css/custom.css]: can you see this like

File: src/css/custom.css [hunk 5]
```diff
@@ -73,6 +73,9 @@
     background: var(--ifm-color-emphasis-200);
     color: inherit;
   }
+  .nav-emoji{
```
Comment (Ajay-Dhangar) [inline src/css/custom.css:76]: can use here...

**Review by Ajay-Dhangar — COMMENTED** [review]
New Conversation for PR...

File: src/css/custom.css [hunk 6]
```diff
@@ -225,4 +230,12 @@ table, tr, td, th {
 
 .hard {
   color: rgb(255 45 85);
+}
+
+/*  for emoji's */
+
+@media screen and (min-width: 1120px) {
+  .nav-emoji{
```
Comment (wingkwong) [inline src/css/custom.css:238]: add a space before `{`

File: src/css/custom.css [hunk 7]
```diff
@@ -225,4 +230,12 @@ table, tr, td, th {
 
 .hard {
   color: rgb(255 45 85);
+}
+
+/*  for emoji's */
```
Comment (wingkwong) [inline src/css/custom.css:235]: remove this comment. below css is self-explanatory already.

File: src/css/custom.css [hunk 8]
```diff
@@ -101,6 +104,8 @@ div[class^='announcementBar_'] {
   font-size: 18px;
   height: 35px;
   padding: 10px;
+  padding-top: 30px;
+  padding-bottom: 30px;
```
Comment (wingkwong) [inline src/css/custom.css:108]: By adding such change, it'll change the style in all views and this solution is not feasible at all.

image

In fact, the correct changes I'm expecting are rather simple. You just need to remove all the paddings (L106 - 108) and set `height` to `auto`.

**Review by wingkwong — CHANGES_REQUESTED** [review]
Final round of changes. After that it's ready for the merge.

File: src/css/custom.css (hunk 8 above)
Comment (Ajay-Dhangar) [reply src/css/custom.css:108]: okay,

Comment (Ajay-Dhangar) [reply src/css/custom.css:108]: Done...

File: src/css/custom.css (hunk 7 above)
Comment (Ajay-Dhangar) [reply src/css/custom.css:233]: okay

**Review by Ajay-Dhangar — COMMENTED** [review]
...

File: src/css/custom.css (hunk 6 above)
Comment (wingkwong) [reply src/css/custom.css:234]: not resolved.

File: src/css/custom.css (hunk 5 above)
Comment (wingkwong) [inline src/css/custom.css:76]: add a space before `{`

File: src/css/custom.css [hunk 9]
```diff
@@ -99,7 +102,7 @@ div[class^='announcementBar_'] {
   border-bottom: 1px dotted #eeeeee;
   font-weight: bold;
   font-size: 18px;
-  height: 35px;
+  height: auto;
   padding: 10px;
```
Comment (wingkwong) [inline src/css/custom.css:106]: this padding needs to be removed.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: src/css/custom.css (hunk 9 above)
Comment (Ajay-Dhangar) [reply src/css/custom.css:106]: okay, done on my side.

Comment (Ajay-Dhangar) [reply src/css/custom.css:106]: okay, done on my side.

Comment (Ajay-Dhangar) [reply src/css/custom.css:106]: okay, done on my side.

**Review by Ajay-Dhangar — COMMENTED** [review]
done my side and can you view any mistake then say me.

**Comment by julix-unity** [comment]
Sorry for commenting on resolved thing, but this was fun to read! Great job being patient, wingkwong and ajay stuck with it and got it done eventually. 

And look at you now! @ajay-dhangar - your profile is wild. Don't think I've ever seen one so jam packed with accomplishments etc. - I'm still new to open source, your sticking-with-it is inspiring. The batch of daily medals too. I didn't even know they were a thing, maybe I'll give that a try.

**Pull request merged**